script/utils/update-redirects-data.ts

```diff
@@ -21,49 +32,66 @@ const updateRedirectsData = async(currentBook: BookWithOSWebData, newBook: BookW
   const redirects: RedirectsData = fs.existsSync(redirectsBookPath) ? await import(redirectsBookPath) : [];
 
   const flatCurrentTree = flattenArchiveTree(currentBook.tree).filter((section) => section.id !== currentBook.id);
-  const flatNewTree = flattenArchiveTree(newBook.tree).filter((section) => section.id !== currentBook.id);
-
-  const formatSection = (section: LinkedArchiveTreeNode) => ({
-    bookId: currentBook.id,
-    pageId: section.id,
-    pathname: decodeURI(
-      content.getUrl({ book: { slug: currentBook.slug }, page: { slug: section.slug } })
-    ),
-  });
+  const currentSections = flatCurrentTree.filter(archiveTreeSectionIsPage);
+  const flatNewTree = flattenArchiveTree(newBook.tree).filter((section) => section.id !== newBook.id);
 
-  const matchRedirect = (section: LinkedArchiveTreeNode) => isEqual(formatSection(section));
+  const matchSlug = (currentPageSlug: string) => flatNewTree.find((newPage) => newPage.slug === currentPageSlug);
+  const matchRedirect = (section: LinkedArchiveTreeNode | ArchiveTreeNode) => isEqual(formatRedirectSource(section));
```

Comment:
we can't use isEqual here anymore, because a whole redirect is being passed in, and we're comparing it to a string. you could do like
```suggestion
  const matchRedirect = (section: LinkedArchiveTreeNode | ArchiveTreeNode) => (redirect: {pathname: string}) => redirect.pathname === formatRedirectSource(section);
```